src/HtmlParser.js

```diff
@@ -0,0 +1,190 @@
+    let lengthSeconds = 0
+    const hoursMinutesSeconds = lengthText.match(/(\d(\d)*)/g)
+    // calculate the time in seconds for every entry
+    for (let i = hoursMinutesSeconds.length - 1; i >= 0; i--) {
+      lengthSeconds += Math.pow(60, (hoursMinutesSeconds.length - i - 1)) * hoursMinutesSeconds[i]
+    }
+    return lengthSeconds
+  }
+
+  // calculates the number of views from the corresponding string "xxx,xxx,xxx,xxx"
+  static calculateViewCount(viewText) {
+    let viewCount = 0
+    const viewersThreeSplit = viewText.match(/(\d(\d)*)/g)
+    for (let i = 0; i < viewersThreeSplit.length; i++) {
+      viewCount = viewCount * 1000 + Number(viewersThreeSplit[i])
+    }
+    return viewCount
+  }
+
+  // calculates the rough timestamp of the release - very exact for minutes, medium exact for hours and loosy exact for days
+  static calculatePublished(publishText, currentTime) {
+    const timePublishedAgo = publishText.match(/(\d(\d)*)/g)
+    let timeSpan
+    if (publishText.indexOf('week') > -1) {
+      // posted x weeks ago
+      timeSpan = Number(timePublishedAgo[0]) * 24 * 360 * 1000 * 7
+    } else if (publishText.indexOf('day') > -1) {
+      // posted x days ago
+      timeSpan = Number(timePublishedAgo[0]) * 24 * 360 * 1000
+    } else if (publishText.indexOf('hour') > -1) {
+      // posted x hours ago
+      timeSpan = Number(timePublishedAgo[0]) * 360 * 1000
+    } else {
+      // posted x minutes ago, just in case
+      timeSpan = Number(timePublishedAgo[0]) * 60 * 1000
+    }
+    return currentTime - timeSpan
+  }
+
+  // creates a list of dictionaries with the relevant data for the different thumbnails
+  // TODO maxres.jpg does not load even tho it loads in invidious
+  static extractThumbnailData(videoId) {
+    // TODO: make customizable
+    return [
+      this.createThumbnailDictionary('maxres', `https://i.ytimg.com/vi/${videoId}/maxres.jpg`, 1280, 720),
+      this.createThumbnailDictionary('maxresdefault', `https://i.ytimg.com/vi/${videoId}/maxresdefault.jpg`, 1280, 720),
+      this.createThumbnailDictionary('sddefault', `https://i.ytimg.com/vi/${videoId}/sddefault.jpg`, 640, 480),
+      this.createThumbnailDictionary('high', `https://i.ytimg.com/vi/${videoId}/hqdefault.jpg`, 480, 360),
+      this.createThumbnailDictionary('medium', `https://i.ytimg.com/vi/${videoId}/mqdefault.jpg`, 320, 180),
+      this.createThumbnailDictionary('default', `https://i.ytimg.com/vi/${videoId}/default.jpg`, 120, 90),
+      this.createThumbnailDictionary('start', `https://i.ytimg.com/vi/${videoId}/1.jpg`, 120, 90),
+      this.createThumbnailDictionary('middle', `https://i.ytimg.com/vi/${videoId}/2.jpg`, 120, 90),
+      this.createThumbnailDictionary('end', `https://i.ytimg.com/vi/${videoId}/3.jpg`, 120, 90),
+    ]
+  }
+
+  static createThumbnailDictionary(Quality, Url, Width, Height) {
+    return {
+      quality: Quality,
+      url: Url,
+      width: Width,
+      height: Height
```

Comment:
As these aren't class names, it's probably a good idea to make these lowercase.
```suggestion
  static createThumbnailDictionary(quality, url, width, height) {
    return {
      quality: quality,
      url: url,
      width: width,
      height: height
```